grpc-api/src/main/java/io/deephaven/grpc_api/table/TableServiceGrpcImpl.java

```diff
@@ -290,10 +290,8 @@ public void exportedTableUpdates(final ExportedTableUpdatesRequest request, fina
     }
 
     public static ExportedTableCreationResponse buildTableCreationResponse(final TableReference tableRef, final Table table) {
-        final String[] columnNames = table.getDefinition().getColumnNamesArray();
-        final ColumnSource<?>[] columnSources = table.getColumnSources().toArray(ColumnSource.ZERO_LENGTH_COLUMN_SOURCE_ARRAY);
         final FlatBufferBuilder builder = new FlatBufferBuilder();
-        builder.finish(BarrageSchemaUtil.makeSchemaPayload(builder, table.getDefinition(), table.getAttributes()));
+        builder.finish(BarrageSchemaUtil.makeSchemaPayload(builder, table));
```

Comment:
let's fix this too, so clients only have one way to decode schemas.

JS API change, in ClientTableState.java, roughly line 983 - this is the current code:
```
    public void applyTableCreationResponse(ExportedTableCreationResponse def) {
        // by definition, the ticket is now exported and connected
        handle.setState(TableTicket.State.EXPORTED);
        handle.setConnected(true);

        Schema schema = Schema.getRootAsSchema(new io.deephaven.javascript.proto.dhinternal.flatbuffers.ByteBuffer(def.getSchemaHeader_asU8()));
        ```
        
This should instead read a Message, confirm there is a Schema inside, read the Schema, and then pass that along. Here's how we read BarrageRecordBatches in that way, from WorkerConnection.java, around line 1030:

                    Message headerMessage = Message.getRootAsMessage(new io.deephaven.javascript.proto.dhinternal.flatbuffers.ByteBuffer(data.getDataHeader_asU8()));
                    if (body.limit() == 0 && headerMessage.headerType() != MessageHeader.BarrageRecordBatch) {
                        // a subscription stream presently ignores schemas and other message types
                        return;
                    }
                    BarrageRecordBatch header = headerMessage.header(new BarrageRecordBatch());

Reply:
good idea: https://github.com/deephaven/deephaven-core/pull/926